File: apps/api/src/main/java/dev/kurama/api/core/hateoas/model/ThemeModel.java
package dev.kurama.api.core.hateoas.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.springframework.hateoas.RepresentationModel;

@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class ThemeModel extends RepresentationModel<ThemeModel> {

  private String primaryColor;
  private String accentColor;
  private String warnColor;
}

File: libs/ui/shared/domain/src/lib/domain/application/theme.model.ts
import { IResource, Resource } from '@hal-form-client';

export interface IThemeModel extends IResource {
  primaryColor?: string;
  accentColor?: string;
  warnColor?: string;
}

export class ThemeModel extends Resource implements IThemeModel {
  primaryColor?: string;
  accentColor?: string;
  warnColor?: string;
}
